### ISSUE TITLE
static_assert is not valid C11

### ISSUE THREAD

**Issue by Honeybunch** [ionescu007/minlzma#1]
static_assert is a C++ keyword and thus this does not compile when using a purely C compiler (clang via LLVM for Windows in my case)

`_Static_assert` is the appropriate keyword in C11.

However MSVC's cl compiler is a C++ compiler and doesn't really implement C11. So pure usage of `_Static_assert` is not valid. 

An important thing to note here is that when checking for MSVC in CMake you cannot purely rely on the MSVC keyword. When using the clang-cl driver for Visual Studio (configured with cmake by passing -T clangcl) CMake will report set MSVC to true since clang-cl is a MSVC-styled compiler. But just because clang-cl is MSVC-styled that doesn't mean it's really 100% compatible. 

clang and clang-cl on Windows work properly with `_Static_assert` since they both compile .c files as C11. Neither will work with `static_assert`.

MS's cl compiler will compile with `static_assert` but will not compile with `_Static_assert`

**Comment by ionescu007** [comment]
Latest `clang` (at least on Linux) seems to compile `static_assert` just fine as long as `assert.h` is added and you use `--std=c11`:

```
ionescu@DESKTOP-TBH09DR:$ clang -c xzstream.c
In file included from xzstream.c:32:
./xzstream.h:67:15: error: expected parameter declarator
static_assert(sizeof(XZ_STREAM_HEADER) == 12, "Invalid Stream Header Size");
ionescu@DESKTOP-TBH09DR$
```

Compare with:
```
ionescu@DESKTOP-TBH09DR$ clang -c xzstream.c --std=c11
ionescu@DESKTOP-TBH09DR$
```

The CMakeFile enforces this:
`set_target_properties(minlz_obj PROPERTIES C_STANDARD 11 C_STANDARD_REQUIRED YES C_EXTENSIONS NO)`

**Comment by Honeybunch** [comment]
You're right. I was incorrect and while `_Static_assert` is a C11 keyword `static_assert` is available in `<assert.h>`.  I misread some docs trying to figure out what was going on. My bad!

However LLVM for Windows does not compile `static_assert` in this project

```
clang -c lzmadec.c -std=c11
lzmadec.c:576:5: error: expected expression
    static_assert((LZMA_BIT_MODEL_SLOTS * 2) == sizeof(g_Decoder.BitModel), "Invalid size");
```

It does compile a simple case:
```
#include <assert.h>
static_assert(sizeof(int) == sizeof(int), "Invalid size");
```

But it doesn't compile a slightly different simple case:
```
#include <assert.h>

int test()
{
    static_assert(sizeof(int) == sizeof(int), "uh");
}
```

```
clang -c test.c -std=c11
test.c:5:5: error: expected expression
    static_assert(sizeof(int) == sizeof(int), "uh");
```

This seems like a very strange compiler bug! It may make sense to move 
```
static_assert((LZMA_BIT_MODEL_SLOTS * 2) == sizeof(g_Decoder.BitModel), "Invalid size");
```
From inside the body of LzInitialize to a higher scope.

**Comment by fingolfin** [comment]
So this library then requires C11? Pity, unfortunately for now I am stuck with C99 (not my choice).

Perhaps this is something that could be pointed in the README?

**Comment by nico** [comment]
Someone reported the clang-cl bug in https://bugs.llvm.org/show_bug.cgi?id=48904 and I just fixed it. The fix will be in LLVM 12.0.

(/me waves at @fingolfin -- hey Max, long time no see :))

**Comment by ionescu007** [comment]
Hi Max,

The library does not "require C11"... an assert is merely a way of
double-checking the correctness of the code. You can comment it out to
work-around the _compiler bug_ you are experiencing, as it is not
C11-specific. You can also move it up higher in scope as suggested.

Let me know if this helps with your issue.

Best regards,
Alex Ionescu
